clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -374,7 +406,9 @@ private void handleListOffsetResponse(TopicPartition topicPartition,
                 }
 
                 long offset = this.subscriptions.fetched(partition);
-                fetch.put(partition, new FetchRequest.PartitionData(offset, this.fetchSize));
+                long consumed = this.subscriptions.consumed(partition);
+                if (consumed == offset)
```

Comment:
Not exactly. This is actually a bug fix. This check is to make initFetch() consistent with handleFetchResponse() -- if consumed != fetched, then the fetched data will be discarded in handleFetchResponse() anyway. Even worse, handleFetchResponse() will reset fetched to consumed, re-fetch the already fetched data again, which is then discarded in fetchedRecords().

Sure, let me rename it to fetched.